### ISSUE TITLE
Gitea looks up a public key with id X in the database, while authorized_keys contains `key-Y`

### ISSUE THREAD

**Issue by anthonyvdotbe** [go-gitea/gitea#11267]
- Gitea version (or commit ref): [1.11.4 arm-6](https://dl.gitea.io/gitea/1.11.4/gitea-1.11.4-linux-arm-6)
- Git version: 2.20.1
- Operating system: Raspbian GNU/Linux 10 (buster)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist: I couldn't get the log to to show anything useful w.r.t. the issue

## Description
My authorized_keys file contains:
```
# gitea public key
command="/usr/local/bin/gitea --config='/etc/gitea/app.ini' serv key-4",no-port-forwarding,no-X11-forwarding,no-agent-forwarding,no-pty ...
```

And the database table `public_key` has a single record with id 4.
However, when I attempt to do a `git push` over SSH, I get [this error](https://github.com/go-gitea/gitea/blob/master/cmd/serv.go#L195), caused by [this error](https://github.com/go-gitea/gitea/blob/41f05588edf8026e43e799dd56114ac001bd7589/routers/private/serv.go#L160):
> Unauthorized
> Cannot find key: 1

After updating the database record's id to 1, the `git push` works, even though the `authorized_keys` file still contains `key-4`.

So it seems that, somewhere between parsing the `key-4` argument, and doing the lookup in the database, the 4 gets converted into 1.

## Screenshots
n/a

**Comment by guillep2k** [comment]
Either that's not the authorized_keys file Gitea is actually using or for some reason (e.g. restore from backup) the file became out of sync with regards to the database. There's a "sync authorized keys" command in the site admin menu you can use to make sure they're in sync.

**Comment by anthonyvdotbe** [comment]
It's neither of these, unfortunately. As explained, the authorized_keys file and the database were in sync, both had id 4, but it didn't work. Only after manually changing the id of the database record to 1, did it work again.

So after your response, here's what I did:
* ran the script "Update the '.ssh/authorized_keys' file with Gitea SSH keys. (Not needed for the built-in SSH server.)". Now both the file and database had id 1 again
* removed, and readded the SSH key. Now both the file and database had id 5. This is the same situation as I had initially with id 4, and SSH is broken at this point because Gitea still looks up id 1
* changed the database record manually again to id 1, leaving the file untouched at id 5. Now SSH works again

If it's not specific to my Gitea instance, this should be easy to reproduce: add an SSH key, remove it, and add the same SSH key again. At this point, both the authorized_keys file and the database should have the same id, which should be a value > 1, but SSH should be broken.

**Comment by guillep2k** [comment]
This still sounds like a configuration problem. Is it possible that the `gitea` process that is running is not the same process that runs when ssh is invoked?

**Comment by guillep2k** [comment]
Sorry, I meant is it the same _executable file_.

**Comment by anthonyvdotbe** [comment]
With `sudo find / -type f -executable -name '*gitea*'`, there's a single result (ignoring the hooks): `/usr/local/bin/gitea`

Doing `ps -F -p $(pgrep gitea)`, the result is:
> UID        PID  PPID  C    SZ   RSS PSR STIME TTY          TIME CMD
> git        447     1  2 250912 89084  3 May01 ?        01:37:38 /usr/local/bin/gitea web --config /etc/gitea/app.ini

So there's a single executable file, and there's a single process that has been running since May 1.

Would you mind trying to reproduce? As far as I know, there's nothing particular about my installation, so I believe it should be doable.

**Comment by zeripath** [comment]
I've just tried this on master, release/v1.11 and v1.11.4 and cannot reproduce.

**Comment by zeripath** [comment]
There is something odd about your authorized_keys I suspect. Are you sure that Gitea is writing to the correct authorized_keys file and that you're sshing to the correct user?

**Comment by anthonyvdotbe** [comment]
My bad, I found the problem... I'm using an SSH certificate which contains the command to use. So the `authorized_keys` file isn't used at all & gitea receives `key-1` because that's what's in the certificate.

My apologies for the inconvenience.
